core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -609,6 +617,27 @@ private class BufferReleasingInputStream(
   override def reset(): Unit = delegate.reset()
 }
 
+/**
+ * A listener to be called at the completion of the ShuffleBlockFetcherIterator
+ * @param data the ShuffleBlockFetcherIterator to process
+ */
+private class ShuffleFetchCompletionListener(var data: ShuffleBlockFetcherIterator)
```

Comment:
Can we move it as an inner class of `ShuffleBlockFetchIterator` and still keep the modifier of `private[this] def cleanup` ?

Just a thought, I am fine with current change.

Reply:
@advancedxy it's ok, I have no idea which is better, so I just put it here. Seems not common to make similar class as an inner class in spark?